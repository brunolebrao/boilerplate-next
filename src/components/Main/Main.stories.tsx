import { Story, Meta } from '@storybook/react/types-6-0'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Boilerplate',
    description: 'TypeScript, ReactJS, NextJS and Styled Components'
  }
} as Meta

export const Basic: Story = (args) => <Main {...args} />
