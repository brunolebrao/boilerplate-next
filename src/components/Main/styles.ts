import styled from 'styled-components'

export const Container = styled.main`
  background-color: ${(props) => props.theme.colors.primary.main};
  color: ${(props) => props.theme.colors.commons.white};
  width: 100%;
  height: 100%;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 3rem;
`
export const Logo = styled.img`
  width: 25rem;
  margin-bottom: 2rem;
`
export const Title = styled.h1`
  font-size: 2.5rem;
`
export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`
export const Illustration = styled.img`
  margin-top: 3rem;
  width: min(30rem, 100%);
`
export const Article = styled.article`
  width: min(40rem, 100%);
  margin-top: 2.5rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
`
export const ImgTechnologies = styled.img`
  width: 5rem;
`
