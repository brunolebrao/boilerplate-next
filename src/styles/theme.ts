export const theme = {
  colors: {
    primary: {
      main: '#06092b'
    },
    commons: {
      black: '#000',
      white: '#fff'
    }
  }
}
