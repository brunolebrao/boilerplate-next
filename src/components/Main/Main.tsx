import * as S from './Main.styles'

const Main = ({
  title = 'Boilerplate',
  description = 'TypeScript, ReactJS, NextJS and Styled Components'
}) => (
  <S.Container>
    <S.Logo src="/img/code.svg" alt="logo" />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Article>
      <S.ImgTechnologies src="/img/ts.svg" alt="logo" />
      <S.ImgTechnologies src="/img/react.svg" alt="logo" />
      <S.ImgTechnologies src="/img/next.svg" alt="logo" />
      <S.ImgTechnologies src="/img/storybook.png" alt="logo" />
      <S.ImgTechnologies src="/img/sc.png" alt="logo" />
    </S.Article>
  </S.Container>
)

export default Main
