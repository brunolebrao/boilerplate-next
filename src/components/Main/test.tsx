import { render, screen } from '@testing-library/react'
import { ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../styles/theme'
import Main from '.'

const renderComponent = (children: ReactNode) =>
  render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

describe('<Main />', () => {
  it('Should render the heading', () => {
    const { container } = renderComponent(<Main />)

    expect(
      screen.getByRole('heading', { name: /Boilerplate/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render the color correctly', () => {
    const { container } = renderComponent(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
