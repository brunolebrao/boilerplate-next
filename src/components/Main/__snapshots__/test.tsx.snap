// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should render the heading 1`] = `
<main>
  <h1>
    React Avan√ßado
  </h1>
</main>
`
